src/librustdoc/html/static/main.js

```diff
@@ -935,37 +935,39 @@
                 var previousTitle = document.title;
 
                 window.onpopstate = function(e) {
-                    var params = getQueryStringParams();
-                    // When browsing back from search results the main page
-                    // visibility must be reset.
-                    if (!params.search) {
-                        var main = document.getElementById('main');
-                        if (hasClass(main, 'content')) {
-                            removeClass(main, 'hidden');
-                        }
-                        var search = document.getElementById('search');
-                        if (hasClass(main, 'content')) {
-                            addClass(main, 'hidden');
+                    if (location.href.indexOf(".html?search=") !== -1) {
+                        var params = getQueryStringParams();
+                        // When browsing back from search results the main page
+                        // visibility must be reset.
+                        if (!params.search) {
+                            var main = document.getElementById('main');
+                            if (hasClass(main, 'content')) {
+                                removeClass(main, 'hidden');
+                            }
+                            var search = document.getElementById('search');
+                            if (hasClass(main, 'content')) {
+                                addClass(main, 'hidden');
+                            }
```

Comment:
This was the issue actually. I thought the variable was limited to the scope, big mistake!